backend/common/src/main/java/ai/verta/modeldb/common/db/migration/Migrator.java

```diff
@@ -218,6 +221,44 @@ void executeSingleMigration(Migration migration) throws IOException, SQLExceptio
     }
   }
 
+  /**
+   * Pre-initializes the schema migrations table, in cases where the database is being converted
+   * from a legacy migration tool (i.e. liquibase).
+   *
+   * <p>_If and only if_ there are legacy migrations in place, there is no existing
+   * schema_migrations table, AND an assumed current version is provided will the schema_versions
+   * table be initialized to the provided value.
+   *
+   * <p>If legacy migrations are in place, and you do not provide an assumed current version, this
+   * method will throw an {@link IllegalStateException}.
+   *
+   * @param legacyMigrationsInPlace Whether legacy migrations have already been applied to this
+   *     database.
+   * @param assumedCurrentVersion The version that the legacy migrations should correspond to.
+   */
+  public void preInitializeIfRequired(
+      boolean legacyMigrationsInPlace, Optional<Integer> assumedCurrentVersion)
+      throws SQLException {
+    if (!legacyMigrationsInPlace) {
+      return;
+    }
+    if (assumedCurrentVersion.isEmpty()) {
+      throw new IllegalStateException(
+          "If legacy migrations are in place, you *must* provide a version to initialize the schema migrations to.");
+    }
+    try {
+      MigrationState currentState = findCurrentState();
+      log.info(
+          "schema_versions table already exists. Skipping pre-initialization step. current state: "
+              + currentState);
+    } catch (SQLException e) {
+      log.info(
+          "No schema_versions table found. Initializing to version " + assumedCurrentVersion.get());
+      setupDatastore();
+      updateVersion(false, assumedCurrentVersion.get());
+    }
```

Comment:
Handling existence like this. could we use `Optional<MigrationState> currentState = findCurrentState();` and instead of retuning null from `findCurrentState();` method we can return Option.

Reply:
sure, that can be done. It won't impact this code path, though, which will throw an exception if the table doesn't exist.

Reply:
we need to be able to distinguish between "the table doesn't exist" and "the table exists, but is empty"

Reply:
I'm planning on doing some refactoring of this code once this is merged, so I'll think about how to improve this when I do that bit of work.